### DISCUSSION TITLE
Setting up Qudi for ESR experiments

### DISCUSSION THREAD

**Discussion by Vinay1786** [Ulm-IQO/qudi-core#29] [Q&A] [answered]
Hi all, 

I installed Qudi software about a year ago for confocal measurements and have been using it happily since then. Now, I would like to set up ESR experiments (ODMR and other standard pulsed measurements) using Qudi and wonder if I can get some guidance here. 

I have a MW source ([N5171B Agilent](https://www.valuetronics.com/pub/media/vti/datasheets/Agilent%20N5171B,%20N5172B%20EXG%20X%20Series.pdf)) and a spin core pulse blaster ([PulseBlasterUSB](https://www.spincore.com/products/PulseBlasterUSB/)), which I am planning to use with Qudi. 

My questions are- 

About my pulse blaster (PB-USB): 

The Qudi hardware module is designed for PB-ESR Pro. Can you kindly brief me what do I need to modify in the hardware module to make it work with PB-USB? 

The clock frequency for PB-USB is 100 MHz (whereas that of PB-ESR Pro is 500 MHz). May I know how does it affect my measurement/if 100 MHz is too low for typical NV experiments? 

According to datasheet of PB-USB: the shortest pulse possible is 50ns. If I’m not wrong, I think the best Rabi pi/2 length I can obtain is 50ns. Apart from that what else might be the disadvantage(s) of having PB-USB? 

About time-tagger:

We are going to buy a new time tagger/fast counter for this set up. I would like to know which model/company (picoquant/swabian/fastcomtec) is recommended for easy and robust interfacing with Qudi.  

 

Sorry if my questions sound too elementary or take a lot of your time. I really need some starting point and I would appreciate any advice on this.  

Thank you very much. 

Vinaya

**Comment by timoML** [comment, marked as answer]
Let me start with what I can say from my personal experience:
- For a pulser 100 MHz of bandwidth is really not optimal. While a 50 ns pi2 pulse is ok for many applications (but especially ones that need to drive all hyperfine levels of the nitrogen may benefit from >10 MHz Rabi drive), the problem is that your granuality of pulse steps is only 10 ns. So if your pi2 is actually 53.2 ns at a given power, you can't accurately cover this. Instead you need to adjust the power to make your 50 ns pulse width match the Rabi drive, which can be painfull. Also, for spectroscopy experiments, the limited phase resolution can cause issues.
- We mainly use fastcomtec (and some swabian) timetaggers. I would expect these modules to be the most robust.

For your other questions: We have some folks with more experience on pulse blaster/ swabian. Let's see whether they can quickly share their view.

**Reply by Vinay1786** [reply]
Hi timoML,

Thanks a lot of the kind reply. I agree with what you mentioned about the pulser. I would still probably give it a go with this pulseblaster and would switch to better one later. We are not planning on multipulse/spectroscopic measurements right now, but only simple coherence/relaxation measurements. Thanks again.

**Comment by LukePiewalker** [comment]
Dear Vinaya,

Regarding my experience with the swabian timetagger, we found it really convenient to get started fast outside of qudi, due to its standalone graphical user interface in a browser and the ability to set up different measurements on the same device, which can also run in parrallel.
The qudi module as it is on main works for standard qudi things if I recall correctly, however there are some specific features implemented which are not as general as it can be (the "sum_channels" feature). Since we use the swabian timetagger as a server on antother pc, our hardware module is slightly diverging from the one in main at the moment.

However, the most used and tested timetagger in qudi is by far the fastcomtec I would say.

**Reply by Vinay1786** [reply]
Thanks Lukas, for the helpful reply. Luckily Qudi has hardware modules for most of the timetaggers, which is a great advantage. Another naive question: are you aware of any instances in Qudi where a particular pulse blaster model has issues with a particular timetagger model? Or, they are just totally independent of each other?

**Reply by liuyanulm** [reply]
Hi Vinay,
They are basically independent of each other. They communicate through TTL signals, which I believe is their only connection. Of course you need to be aware of their resolution and granularities at time scale, when you setup up a pulsed measurement of the spin states.
Yan

**Reply by liuyanulm** [reply]
More specifically, the timetagger is triggered by the pulse blaster with TTL.

**Comment by Vinay1786** [comment]
Hi liuyanulm and everyone,

Thanks for the kind response. As we are trying to interface different hardware, we have come across some fundamental questions that we hope are good enough to be discussed here.

In the hardware module 'fast_counter_fpga_qo.py' :

(1)  Why is the variable '_internal_clock_hz' is being set to 950 MHz even though the clock oscillator frequency is only 100 MHz for the FPGA version (Opal Kelly XEM6310) for which this code is written?

(2) For pulsed measurements, does the use of this FPGA hardware module ('fast_counter_fpga_qo.py') require an external clock/fast counter like the NI card?   (the reason we are asking this question is because in a function that collects the hardware constraints, the code documentation says the function gets the 'hardware constraints of the fast counting device'. Since you mention about getting constraints from a fast counting device, I wonder whether you are just referring to the FPGA itself as a fast counting device or some other external hardware like NI card.)

If you could comment on these points that would be great!

Thanks,
Vinaya

**Reply by Neverhorst** [reply]
Hi @Vinay1786,
I'm happy to answer this since I wrote the module.
Can you please ask this question in a new thread though since your original question has been answered already? 
Thank you.

**Reply by Vinay1786** [reply]
Oops, I am sorry! Thanks, Nikolas for pointing this out! I have moved the question to a new thread.

Regards,
Vinaya
